presto-parser/src/main/java/com/facebook/presto/sql/tree/ShowStats.java

```diff
@@ -13,30 +13,48 @@
  */
 package com.facebook.presto.sql.tree;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
 import java.util.Objects;
 import java.util.Optional;
 
+import static com.facebook.presto.sql.QueryUtil.simpleQuery;
 import static com.google.common.base.MoreObjects.toStringHelper;
 
 public class ShowStats
         extends Statement
 {
-    private final QualifiedName table;
+    private final Query query;
 
+    @VisibleForTesting
     public ShowStats(QualifiedName table)
     {
         this(Optional.empty(), table);
     }
 
-    public ShowStats(NodeLocation location, QualifiedName table)
+    @VisibleForTesting
+    public ShowStats(Query query)
+    {
+        this(Optional.empty(), query);
+    }
+
+    public ShowStats(Optional<NodeLocation> location, QualifiedName table)
     {
-        this(Optional.of(location), table);
+        this(location, createFakeSpecification(location, table));
     }
 
-    private ShowStats(Optional<NodeLocation> location, QualifiedName table)
+    public ShowStats(Optional<NodeLocation> location, Query query)
     {
         super(location);
-        this.table = table;
+        this.query = query;
+    }
+
+    private static Query createFakeSpecification(Optional<NodeLocation> location, QualifiedName name)
```

Comment:
This should be called createFakeQuery and it's used to handle two syntaxes using only one rewrite. We create fake query for `SHOW STATS qualifiedName` case as there is no query there.
Creating this query may be moved to AstBuilder if you like that more.